### ISSUE TITLE
Fit does not go to 0 at end of series

### ISSUE THREAD

**Issue by cschwarz-stat-sfu-ca** [cschwarz-stat-sfu-ca/BTSPAS#31]
# This was moved from Issue #30 to create new issue.

 However, I am running into another issue with the data set I previously shared and was curious if you have any insight.

In short, the model is having issues converging on a "believable" estimate when I specify maiden catch (u2) as "bad" for 1-2 periods near the end of the time series (e.g., bad.u2 <-c(10,11)). For context, I want to identify these periods as bad because streamflow got so low that the trap essentially stopped spinning and yearling migrants can then avoid the trap.

When I run the model without specifying periods 10 and/or 11 as bad, the model estimates abundance declining towards zero in the final periods, which is what we would expect. However, when I specify u2 in periods 10 and/or 11 as bad, the model estimates that abundance increases quite dramatically towards the tail end of the time series (and the precision blows up). I've run this model with up to 500K iterations/250 burn-in per chain (4 chains) and still have issues. Looking at the model outputs, there are clearly some issues (e.g., estimates of logitP >5000 and < -5000 - agh!!).

This specific time series is pretty short (only 13 periods) and the 1-2 bad periods are right after peak outmigration both of which perhaps are contributing to the estimation problems. To troubleshoot, I toyed around with adding additional "fake periods" of zero catch with the idea of helping the model realize catch should decline to zero but this doesn't seem to help.

Here's the problematic dataset: table_Run5_Final_Catch_Summary.csv
[table_Run5_Final_Catch_Summary.csv](https://github.com/cschwarz-stat-sfu-ca/BTSPAS/files/7530083/table_Run5_Final_Catch_Summary.csv)

**Comment by kalebentley** [comment]
Hi Carl,

I looked through your response a couple of weeks ago but have been slow to respond as I am still having issues with the data set I shared with you before (reattached here - [table_Run5_Final_Catch_Summary.csv](https://github.com/cschwarz-stat-sfu-ca/BTSPAS/files/7620699/table_Run5_Final_Catch_Summary.csv))

In short, I've tried the "1" trick you suggested but I am still getting wonky results with the non-diagonal model (i.e., the estimates of abundance and capture probability will not converge).  Specifically, at least one of the chains ultimately does not mix for despite trying up to 1 M iterations with 900K burn-in per chain and I end up with <50 effective draws.  I've never run into this issue before.   

Interestingly, the diagonal model converges but there are definitely some delayed recaptures and the total estimate of abundance is quite a bit different between the two models (assuming the non-diagonal total estimate is close to "truth").  

One obvious thing about this data set is that the sample sizes for non-diagonal recaptures are pretty small and limited to just a few periods.  While this pattern isn't atypical at our traps, and I've never had this sort of issue before, out of curiosity I manually edited the data set to combine off-diagonal recaptures from the "plus 2" and "plus 3" periods.  Unfortunately, this did not help.  I tried other tricks like this trying to find the "weak link" but no luck.

Again, I've never run into this issue before with the non-diagonal model and so it's a bit puzzling.  Any additional thoughts or suggestions would be greatly appreciated.  

Kale
